### PR TITLE
minor refactoring + client SSL test

### PR DESCRIPTION
**Pull request by Sammers21**
Changes:
* MqttConnection was renamed to MqttServerConnection. I think it is better because we have MqttCleintConnection for a client.

* setTrustAll method in MqttClientOptions was overridden

* SSL test

### CODE REVIEW

File: src/test/java/io/vertx/mqtt/test/client/MqttClientSslTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      .setPort(MQTT_SERVER_TLS_PORT)
+      .setHost(MQTT_SERVER_HOST)
+      .setKeyCertOptions(pemKeyCertOptions)
+      .setSsl(true);
+
+    MqttClientOptions clientOptions = new MqttClientOptions()
+      .setPort(MQTT_SERVER_TLS_PORT)
+      .setHost(MQTT_SERVER_HOST)
+      .setSsl(true)
+      .setTrustAll(true);
```
Comment (ppatierno) [inline src/test/java/io/vertx/mqtt/test/client/MqttClientSslTest.java]: Can you add another test where you don't use "trust all" but use the client-truststore we have in the resources ? Just to confirm that the handshake is going well with the server certificate. Finally, did you take a look at the network traffic with Wireshark just to have another way to check the SSL/TLS handshake ?

Comment (Sammers21) [inline src/test/java/io/vertx/mqtt/test/client/MqttClientSslTest.java]: image

Comment (Sammers21) [inline src/test/java/io/vertx/mqtt/test/client/MqttClientSslTest.java]: this is what Wireshark shows when running just added "clientSslClientTruststoreTest" test

Comment (Sammers21) [inline src/test/java/io/vertx/mqtt/test/client/MqttClientSslTest.java]: Btw, Wireshark is a great tool.

File: src/main/java/io/vertx/mqtt/MqttClientOptions.java [hunk 2]
```diff
@@ -272,6 +273,17 @@ public MqttClientOptions setSsl(boolean ssl) {
...
+  public MqttClientOptions setTrustStoreOptions(JksOptions options) {
+    super.setTrustStoreOptions(options);
+    return this;
+  }
+
+  @Override
+  public MqttClientOptions setTrustAll(boolean trustAll) {
+    super.setTrustAll(trustAll);
+    return this;
+  }
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/MqttClientOptions.java]: As already provided by the server component, even if without tests, we should add even the following ways to provide certs (as PEM, PFX, ...) overriding following methods : 

`setKeyCertOptions`
`setKeyStoreOptions`
`setPfxKeyCertOptions`
`setPemKeyCertOptions`
`setTrustOptions`
`setPemTrustOptions`
`setPfxTrustOptions`

and even adding following override to be more complete : 

`addEnabledCipherSuite`
`addEnabledSecureTransportProtocol`
`addCrlPath`
`addCrlValue`

File: src/test/java/io/vertx/mqtt/test/client/MqttClientSslTest.java (hunk 1 above)
Comment (ppatierno) [inline src/test/java/io/vertx/mqtt/test/client/MqttClientSslTest.java]: I know, it's my preferred tool :-)

File: src/main/java/io/vertx/mqtt/MqttClientOptions.java (hunk 2 above)
Comment (Sammers21) [inline src/main/java/io/vertx/mqtt/MqttClientOptions.java]: Done

**Comment by ppatierno** [comment]
@Sammers21 I'm good to go but you should squash :-)

**Comment by Sammers21** [comment]
@ppatierno, squashed

**Comment by ppatierno** [comment]
@Sammers21 thanks ! I'm going to merge it !

**Pull request merged**